include/boost/url/impl/url_base.ipp

```diff
@@ -396,20 +396,20 @@ set_password(string_view s)
 
 url_base&
 url_base::
-set_password(pct_encoded_view s)
+set_password(decode_view s)
 {
     grammar::detail::copied_strings<
         BOOST_URL_STACK_BYTES> buf(
             this->string());
     s = s.maybe_copy(buf);
     check_invariants();
     auto const n =
-        detail::pct_encode_bytes_impl(
+        detail::encode_bytes_impl(
```

Comment:
`encoded_size_impl` but why do we need our own private function why can't we call `encoded_size`?

Reply:
`s` is not a `string_view`. We could have other overloads taking iterators but we haven't discussed that yet.